src/vm/eventpipeevent.cpp

```diff
@@ -156,16 +156,15 @@ unsigned int EventPipeEvent::GetMetadataLength() const
 void EventPipeEvent::RefreshState()
 {
     LIMITED_METHOD_CONTRACT;
-    m_enabled = m_pProvider->EventEnabled(m_keywords, m_level);
+    _ASSERTE(EventPipe::IsLockOwnedByCurrentThread());
+    m_enabledMask = m_pProvider->ComputeEventEnabledMask(m_keywords, m_level);
 }
 
 bool EventPipeEvent::IsEnabled(uint64_t sessionMask) const
 {
     LIMITED_METHOD_CONTRACT;
-    _ASSERT(m_pProvider != nullptr);
-    const bool IsProviderEnabled = m_pProvider->IsEnabled(sessionMask);
-    const bool IsEventEnabled = m_pProvider->EventEnabled(m_keywords, m_level);
-    return (IsProviderEnabled && IsEventEnabled);
+    _ASSERTE(m_pProvider != nullptr);
```

Comment:
If this is ever `nullptr`, then we could AV. Should we re-write it:
```cpp
PRECONDITION(m_pProvider != nullptr);
return m_pProvider ? (m_pProvider->IsEnabled(sessionMask) && (m_enabledMask & sessionMask) != 0) : false;
```

Reply:
I do not believe it is possible for `m_pProvider` to be null, it is constructed with an address of a reference and never changes. I don't know if it could become a dangling reference though - I am counting on your change to leak the objects to make sure it stays alive throughout.

Reply:
agreed ... it's more about "preventing" or "future proof" ... it's a pointer after all, for which we are asserting, and do not own... just a preference / nit